x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/AbstractSqlWireSerializingTestCase.java

```diff
@@ -21,12 +22,14 @@
 public abstract class AbstractSqlWireSerializingTestCase<T extends Writeable> extends AbstractWireTestCase<T> {
 
     @Override
-    protected T copyInstance(T instance, Version version) throws IOException {
+    protected T copyInstance(T instance, TransportVersion version) throws IOException {
+        Version nodeVersion = Version.fromId(version.id);
```

Comment:
Sql classes are maintaining Version, so we need to convert back here

Reply:
I wonder if they should not be maintaining a TransportVersion now? It looks like the usage is that only a node with the same version can read the stream https://github.com/elastic/elasticsearch/blob/98fdf4fa59eadac53fffcf9850cce9c733706ece/x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/common/io/SqlStreamInput.java#L33

any views on this @Luegg ?

Reply:
hmm github tricked me and it was already discussed.. https://github.com/elastic/elasticsearch/pull/93384#discussion_r1092113914